routes/item/[id].tsx

```diff
@@ -12,53 +13,66 @@ import {
 import {
   type Comment,
   createComment,
+  getAreVotedBySessionId,
   getCommentsByItem,
   getItemById,
   getManyUsers,
   getUserById,
   getUserBySessionId,
-  getVotedItemsByUser,
   type Item,
   type User,
 } from "@/utils/db.ts";
 import { redirect } from "@/utils/http.ts";
 import UserPostedAt from "@/components/UserPostedAt.tsx";
 import { pluralize } from "@/utils/display.ts";
 
+const PAGE_LENGTH = 10;
+
 interface ItemPageData extends State {
   user: User;
   item: Item;
   comments: Comment[];
   commentsUsers: User[];
   isVoted: boolean;
+  lastPage: number;
+}
+
+function calcPageNum(url: URL) {
+  return parseInt(url.searchParams.get("page") || "1");
+}
+
+function calcLastPage(total = 0, pageLength = PAGE_LENGTH): number {
+  return Math.ceil(total / pageLength);
 }
```

Comment:
Can we centralise and de-duplicate any pagination-related code in the utils folder with accompanying tests?